src/VersionManager.cs

```diff
@@ -19,68 +20,166 @@ internal class VersionManager
+        /// </summary>
+        /// <param name="version">The version to be installed</param>
+        /// <param name="tier">The tier</param>
+        /// <param name="platform">The platform</param>
+        /// <returns>True if installed successfully otherwise false</returns>
+        public static bool DownloadVersion(Version version, ServerTier tier, ServerPlatform platform)
+        {
+            string fullPath = Path.Combine(USER_DR_DIR, "installed", tier.ToString().ToLower(), platform.ToString().ToLower(), version.ToString());
 
-                string stagingPath = Path.Combine(USER_DR_DIR, "Download.zip");
+            string stagingPath = Path.Combine(USER_DR_DIR, "Download.zip");
 
-                string uri = $"https://www.darkriftnetworking.com/DarkRift2/Releases/{version}/{tier}/{platform}/";
-                if (tier == ServerTier.Pro)
+            string uri = $"https://www.darkriftnetworking.com/DarkRift2/Releases/{version}/{tier}/{platform}/";
+            if (tier == ServerTier.Pro)
+            {
+                string invoiceNumber = GetInvoiceNumber();
+                if (invoiceNumber == null)
                 {
-                    string invoiceNumber = GetInvoiceNumber();
-                    if (invoiceNumber == null)
-                    {
-                        Console.Error.WriteLine(Output.Red($"You must provide an invoice number in order to download Pro DarkRift releases."));
-                        return null;
-                    }
-
-                    uri += $"?invoice={invoiceNumber}";
+                    Console.Error.WriteLine(Output.Red($"You must provide an invoice number in order to download Pro DarkRift releases."));
+                    return false;
                 }
 
-                try
+                uri += $"?invoice={invoiceNumber}";
+            }
+
+            try
+            {
+                using (WebClient myWebClient = new WebClient())
                 {
-                    using (WebClient myWebClient = new WebClient())
-                    {
-                        myWebClient.DownloadFile(uri, stagingPath);
-                    }
+                    myWebClient.DownloadFile(uri, stagingPath);
                 }
-                catch (WebException e)
+            }
+            catch (WebException e)
+            {
+                Console.Error.WriteLine(Output.Red($"Could not download DarkRift {version} - {tier} (.NET {platform}):\n\t{e.Message}"));
+                return false;
+            }
+
+            Console.WriteLine($"Extracting package...");
+
+            Directory.CreateDirectory(fullPath);
+
+            ZipFile.ExtractToDirectory(stagingPath, fullPath, true);
+
+            Console.WriteLine(Output.Green($"Successfully downloaded package."));
+
+            return true;
+        }
+
+        /// <summary>
+        /// Checks if a version of Dark Rift is installed
+        /// </summary>
+        /// <param name="version"></param>
```

Comment:
These are not filled in.

Also DarkRift is spelt without a space in the summary!